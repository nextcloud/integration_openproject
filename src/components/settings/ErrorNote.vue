<!--
  - SPDX-FileCopyrightText: 2025 Jankari Tech Pvt. Ltd.
  - SPDX-License-Identifier: AGPL-3.0-or-later
-->
<template>
	<NcNoteCard class="note-card" type="error">
		<p class="note-card--title">
			<b>{{ errorTitle }}</b>
		</p>
		<p class="note-card--error-description">
			<span v-if="errorMessage">{{ errorMessage }}. </span><a v-if="errorLink"
				class="link"
				:href="errorLink"
				target="_blank"
				:title="errorLinkLabel">{{ errorLinkLabel }}</a>
		</p>
	</NcNoteCard>
</template>

<script>
import { NcNoteCard } from '@nextcloud/vue'

export default {
	name: 'ErrorNote',
	components: {
		NcNoteCard,
	},
	props: {
		errorTitle: {
			type: String,
			required: true,
		},
		errorMessage: {
			type: String,
			default: '',
		},
		errorLink: {
			type: String,
			default: '',
		},
		errorLinkLabel: {
			type: String,
			default: 'Link',
		},
	},
}
</script>

<style lang="scss">
.note-card {
	max-width: 900px;
}

.link {
	color: #1a67a3 !important;
	font-style: normal;
}
</style>
