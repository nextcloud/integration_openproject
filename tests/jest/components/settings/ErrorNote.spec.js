/**
 * SPDX-FileCopyrightText: 2025 Jankari Tech Pvt. Ltd.
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

import { shallowMount } from '@vue/test-utils'

import ErrorNote from '../../../../src/components/settings/ErrorNote.vue'

describe('Component: ErrorNote', () => {
	it('should show error title', () => {
		const wrapper = getWrapper({ errorTitle: 'Test error' })

		expect(wrapper.element).toMatchSnapshot()
	})
	it('should show error title and error message', () => {
		const wrapper = getWrapper({ errorTitle: 'Test error', errorMessage: 'Test error message' })

		expect(wrapper.element).toMatchSnapshot()
	})
	it('should show all provided props', () => {
		const wrapper = getWrapper({
			errorTitle: 'Test error',
			errorMessage: 'Test error message',
			errorLink: 'http://example.com',
			errorLinkLabel: 'Test link',
		})

		expect(wrapper.element).toMatchSnapshot()
	})
})

function getWrapper(propsData = {}) {
	return shallowMount(ErrorNote, {
		propsData,
	})
}
