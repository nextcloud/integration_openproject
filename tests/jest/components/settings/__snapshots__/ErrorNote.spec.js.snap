// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ErrorNote should show all provided props 1`] = `
<div
  class="notecard note-card notecard--error"
  data-v-722d543a=""
  role="alert"
>
  <span
    aria-hidden="true"
    class="material-design-icon alert-decagram-icon notecard__icon"
    data-v-722d543a=""
    role="img"
  >
    <svg
      class="material-design-icon__svg"
      fill="var(--color-error)"
      height="24"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M23,12L20.56,9.22L20.9,5.54L17.29,4.72L15.4,1.54L12,3L8.6,1.54L6.71,4.72L3.1,5.53L3.44,9.21L1,12L3.44,14.78L3.1,18.47L6.71,19.29L8.6,22.47L12,21L15.4,22.46L17.29,19.28L20.9,18.46L20.56,14.78L23,12M13,17H11V15H13V17M13,13H11V7H13V13Z"
      >
        <!---->
      </path>
    </svg>
  </span>
  <div
    data-v-722d543a=""
  >
    <!---->
    <p
      class="note-card--title"
      data-v-722d543a=""
    >
      <b
        data-v-722d543a=""
      >
        Test error
      </b>
    </p>
     
    <p
      class="note-card--error-description"
      data-v-722d543a=""
    >
      <span
        data-v-722d543a=""
      >
        Test error message. 
      </span>
      <a
        class="link"
        data-v-722d543a=""
        href="http://example.com"
        target="_blank"
        title="Test link"
      >
        Test link
      </a>
    </p>
  </div>
</div>
`;

exports[`Component: ErrorNote should show error title 1`] = `
<div
  class="notecard note-card notecard--error"
  data-v-722d543a=""
  role="alert"
>
  <span
    aria-hidden="true"
    class="material-design-icon alert-decagram-icon notecard__icon"
    data-v-722d543a=""
    role="img"
  >
    <svg
      class="material-design-icon__svg"
      fill="var(--color-error)"
      height="24"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M23,12L20.56,9.22L20.9,5.54L17.29,4.72L15.4,1.54L12,3L8.6,1.54L6.71,4.72L3.1,5.53L3.44,9.21L1,12L3.44,14.78L3.1,18.47L6.71,19.29L8.6,22.47L12,21L15.4,22.46L17.29,19.28L20.9,18.46L20.56,14.78L23,12M13,17H11V15H13V17M13,13H11V7H13V13Z"
      >
        <!---->
      </path>
    </svg>
  </span>
  <div
    data-v-722d543a=""
  >
    <!---->
    <p
      class="note-card--title"
      data-v-722d543a=""
    >
      <b
        data-v-722d543a=""
      >
        Test error
      </b>
    </p>
     
    <p
      class="note-card--error-description"
      data-v-722d543a=""
    >
      <!---->
      <!---->
    </p>
  </div>
</div>
`;

exports[`Component: ErrorNote should show error title and error message 1`] = `
<div
  class="notecard note-card notecard--error"
  data-v-722d543a=""
  role="alert"
>
  <span
    aria-hidden="true"
    class="material-design-icon alert-decagram-icon notecard__icon"
    data-v-722d543a=""
    role="img"
  >
    <svg
      class="material-design-icon__svg"
      fill="var(--color-error)"
      height="24"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M23,12L20.56,9.22L20.9,5.54L17.29,4.72L15.4,1.54L12,3L8.6,1.54L6.71,4.72L3.1,5.53L3.44,9.21L1,12L3.44,14.78L3.1,18.47L6.71,19.29L8.6,22.47L12,21L15.4,22.46L17.29,19.28L20.9,18.46L20.56,14.78L23,12M13,17H11V15H13V17M13,13H11V7H13V13Z"
      >
        <!---->
      </path>
    </svg>
  </span>
  <div
    data-v-722d543a=""
  >
    <!---->
    <p
      class="note-card--title"
      data-v-722d543a=""
    >
      <b
        data-v-722d543a=""
      >
        Test error
      </b>
    </p>
     
    <p
      class="note-card--error-description"
      data-v-722d543a=""
    >
      <span
        data-v-722d543a=""
      >
        Test error message. 
      </span>
      <!---->
    </p>
  </div>
</div>
`;
